### PR TITLE
Backend 33 satisfaction algorithim

### LINKED ISSUE #33: Design algorithm for deciding total user satisfaction
Users are going to submit their current satisfaction with the current temperature. How this question is asked should be irrelevant to the general algorithm. It needs to be able to do the following. 

- Retrieve the user records from where they're stored (Memory for now)
- Combine the satisfaction values to get an average or total across all users
- Decide on the "critical mass" required before alerting maintenance
- Trigger the maintenance alert  and provide it with some detail about the vote (handled by #38)

Time estimate 2-3 days

### PR DESCRIPTION
**Pull request by AvalonXD**
fixes #33 
This should be adequate for the basic problem idea.

I would like some look over as I'm not sure though.

Some of the variables will need to be initialised first especially the new object lists and voteStore may need to be changed but I'm not adept in all of UNIX time yet so don't want to wreck anything.

### CODE REVIEW

File: backend/satisfactionCalculator.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    tempArray[i] = voteStore.keys[i]
+                }
+            }
+            let critMass = 0;
+            for(let i = 0; i < tempArray.length; i++) {
+                if (tempArray[i].timeStamp == timebucket('h').subtract(1) + '') {
+                    critMass++
+                }
+            }
+            if(areaList[i].size/2 == critMass) {
```
Comment (MystikHub) [inline backend/satisfactionCalculator.js]: What does `i` represent here?

Comment (MystikHub) [reply backend/satisfactionCalculator.js]: Also, isn't `areaList` an array of the strings of the area names? Wouldn't `areaList[i].size` return the characters in that particular area name?

Comment (AvalonXD) [reply backend/satisfactionCalculator.js]: You're right it should be areaList[j] and the .size is referring to the number of people within the area. Just a stopover for the critical mass when half of the room votes within the last hour.

**Comment by henrym2** [comment]
It all looks good to me. I'll write some unit tests for these this week.

**Pull request merged**